manifests/init.pp

```diff
@@ -1,60 +1,424 @@
 # == Class: sssd
 #
 # This class allows you to install and configure SSSD.
-#
 # It will forcefully disable nscd which consequently prevents you from using an
 # nscd module at the same time, which is the correct behavior.  ---
 #
+# == Authors
+#
+# @author Trevor Vaughan <mailto:tvaughan@onyxpoint.com>
+#
 # == Parameters:
 #
-# @param pki
-#   * If 'simp', include SIMP's pki module and use pki::copy to manage
-#     application certs in /etc/pki/simp_apps/sssd/x509
-#   * If true, do *not* include SIMP's pki module, but still use pki::copy
-#     to manage certs in /etc/pki/simp_apps/sssd/x509
-#   * If false, do not include SIMP's pki module and do not use pki::copy
-#     to manage certs.  You will need to appropriately assign a subset of:
-#     * app_pki_dir
-#     * app_pki_key
-#     * app_pki_cert
-#     * app_pki_ca
-#     * app_pki_ca_dir
-#
-# @param app_pki_external_source
-#   * If pki = 'simp' or true, this is the directory from which certs will be
-#     copied, via pki::copy.  Defaults to /etc/pki/simp/x509.
-#
-#   * If pki = false, this variable has no effect.
-#
-# @param app_pki_dir
-#   This variable controls the basepath of $app_pki_key, $app_pki_cert,
-#   $app_pki_ca, $app_pki_ca_dir, and $app_pki_crl.
-#   It defaults to /etc/pki/simp_apps/sssd/x509.
+# === Installation / Configuration Parameters ===
+# The following parameters allow you to control the installation and
+# configuration of the sssd package.
 #
-# == Authors
+# @param client_package_name [Optional[String]]
+#   The name of the sssd client package (varies by OS)
+#   Default value: 'sssd-client'
```

Comment:
That makes sense, see other comment. Might be the case in the new puppet-strings that it will pull defaults in from code / hiera, in which case specifying default value in docs would be excessive.